dbfit-java/informix/src/main/java/dbfit/environment/InformixEnvironment.java

```diff
@@ -0,0 +1,321 @@
+        String[] qualifiers = NameNormaliser.normaliseName(tableOrViewName).split("\\.");
+        return readIntoParams(qualifiers, columnsOrParamtersQueryText(false, tableOrViewName));
+    }
+
+    private Map<String, DbParameterAccessor> readIntoParams(String[] queryParameters, String query) throws SQLException {
+        try (PreparedStatement dc = currentConnection.prepareStatement(query)) {
+            for (int i = 0; i < queryParameters.length; i++) {
+                dc.setString(i + 1, NameNormaliser.normaliseName(queryParameters[i]));
+            }
+
+            ResultSet rs = dc.executeQuery();
+            Map<String, DbParameterAccessor> allParams = new HashMap<String, DbParameterAccessor>();
+
+            while (rs.next()) {
+                String paramName = defaultIfNull(rs.getString(1), "");
+                String dataType = rs.getString(2);
+                String direction = rs.getString(3);
+                int position = rs.getInt(4);
+                Direction paramDirection = getParameterDirection(direction);
+                if (paramDirection != RETURN_VALUE && paramName.equals("")) {
+                    throw new SQLException("Missing column or procedure parameter name");
+                }
+                DbParameterAccessor dbp = new DbParameterAccessor(paramName,
+                        paramDirection, getSqlType(dataType),
+                        getJavaClass(dataType),
+                        paramDirection == RETURN_VALUE ? -1
+                                : position);
+                allParams.put(NameNormaliser.normaliseName(paramName), dbp);
+            }
+            rs.close();
+            return allParams;
+        }
+    }
+
+    private static Direction getParameterDirection(int isOutput, String name) {
+        if (name.isEmpty()) {
+            return RETURN_VALUE;
+        }
+        return (isOutput == 1) ? OUTPUT : INPUT;
+    }
+
+    private static Direction getParameterDirection(String direction) {
+        /*  0 = Parameter is of unknown type
+            1 = Parameter is INPUT mode
+            2 = Parameter is INOUT mode
+            3 = Parameter is multiple return value
+            4 = Parameter is OUT mode
+            5 = Parameter is a return value
+        */
+        if ("1".equals(direction))
+            return INPUT;
+        if ("2".equals(direction))
+            return INPUT_OUTPUT;
+        if ("3".equals(direction))
+            return RETURN_VALUE;
+        if ("4".equals(direction))
+            return OUTPUT;
+        if ("5".equals(direction))
+            return RETURN_VALUE;
+        throw new UnsupportedOperationException("Direction " + direction
+                + " is not supported");
+    }
+
+    // List interface has sequential search, so using list instead of array to map types.
+    private static List<String> stringTypes = Arrays.asList(new String[] {
+            "VARCHAR", "LVARCHAR", "CHAR", "TEXT", "NCHAR", "NVARCHAR" });
```

Comment:
In all these the `new String[] {}` seems superfluous. It could be simply `Arrays.asList("VARCHAR", ..., "NVARCHAR")`.